src/test/java/net/barbux/creatures2d/CreatureTest.java

```diff
@@ -0,0 +1,146 @@
+    final Creature.NodeList creatureNodeList = creature.new NodeList();
+    final Creature.Node node = new Creature.Node(2.0, 2.0);
+    Assert.assertTrue(creatureNodeList.add(node));
+  }
+
+  @Test
+  public void cloneOutputNotNull() {
+    final Creature creature = new Creature();
+    final Creature actual = creature.clone();
+    Assert.assertNotNull(actual);
+    final ArrayList<Creature.Bone> arrayList = new ArrayList<Creature.Bone>();
+    Assert.assertEquals(arrayList, actual.allBones);
+    Assert.assertNotNull(actual.allNodes);
+    Assert.assertNull(actual.physics);
+    Assert.assertEquals(0.0, actual.maxReached, 0.0);
+    final ArrayList<Creature.Muscle> arrayList1 = new ArrayList<Creature.Muscle>();
+    Assert.assertEquals(arrayList1, actual.allMuscles);
+    Assert.assertEquals(0, actual.creatureId);
+  }
+
+  @Test
+  public void getCenterOutputNotNull() {
+    final Creature creature = new Creature();
+    final Vector actual = creature.getCenter();
+    Assert.assertNotNull(actual);
+    Assert.assertEquals(0x1.fffffffffffffp+1022 /* 8.98847e+307 */, actual.y, 0.0);
+    Assert.assertEquals(0x1.fffffffffffffp+1022 /* 8.98847e+307 */, actual.x, 0.0);
+  }
+
+  @Test
+  public void getFitnessOutputZero() {
+    final Creature creature = new Creature();
+    Assert.assertEquals(0.0, creature.getFitness(), 0.0);
+  }
+
+  @Test
+  public void getInputPositiveOutputIndexOutOfBoundsException() {
+    final Creature creature = new Creature();
+    final Creature.NodeList creatureNodeList = creature.new NodeList();
+    thrown.expect(IndexOutOfBoundsException.class);
+    creatureNodeList.get(3);
+  }
+
+  @Test
+  public void removeInputPositiveOutputIndexOutOfBoundsException() {
+    final Creature creature = new Creature();
+    final Creature.NodeList creatureNodeList = creature.new NodeList();
+    thrown.expect(IndexOutOfBoundsException.class);
+    creatureNodeList.remove(2);
+  }
+
+  @Test
+  public void sizeOutputZero() {
+    final Creature creature = new Creature();
+    final Creature.NodeList creatureNodeList = creature.new NodeList();
+    Assert.assertEquals(0, creatureNodeList.size());
+  }
+
+  @Test
+  public void cloneOutputNotNull1() {
+    final Creature.Node thisObj = new Creature.Node(0.05, 0.05);
+    final Creature.Node actual = thisObj.clone();
+    Assert.assertNotNull(actual);
+    Assert.assertNotNull(actual.p);
+    Assert.assertEquals(0.05, actual.p.x, 0.0);
+    Assert.assertEquals(0.05, actual.p.y, 0.0);
+    Assert.assertNotNull(actual.v);
+    Assert.assertEquals(0.0, actual.v.x, 0.0);
+    Assert.assertEquals(0.0, actual.v.y, 0.0);
+    Assert.assertNotNull(actual.originalPos);
+    Assert.assertEquals(0.05, actual.originalPos.x, 0.0);
+    Assert.assertEquals(0.05, actual.originalPos.y, 0.0);
+    Assert.assertEquals(0.05, actual.size, 0.0);
+    Assert.assertNotNull(actual.color);
+    Assert.assertEquals(1.0, actual.weight, 0.0);
+    Assert.assertEquals(0, actual.nodeId);
+  }
```

Comment:
This test tests something useful. It's a bit too verbose though. If I really wanted to test that I would introduce an `equals` method on `Node` and use that to compare, rather that asserting the fields.
The other fields (e.g. `color`) are not properly asserted.
:headphones: